pdns/dnsdist-lua-actions.cc

```diff
@@ -776,10 +776,18 @@ DNSAction::Action SpoofAction::operator()(DNSQuestion* dq, std::string* ruleresu
   // do we even have a response?
   if (d_cname.empty() &&
       d_rawResponses.empty() &&
+      // make sure pre-forged response is greater than sizeof(dnsheader)
+      (d_raw.size() < sizeof(dnsheader)) &&
       d_types.count(qtype) == 0) {
     return Action::None;
   }
 
+  if (d_raw.size() >= sizeof(dnsheader)) {
+    auto id = dq->getHeader()->id;
+    dq->getMutableData() = std::move(d_raw);
```

Comment:
The content of `d_raw` is not really moved because the method is `const`, but it would be a problem for the next query if it was ;-)
```suggestion
    dq->getMutableData() = d_raw;
```